checker/src/main/java/org/checkerframework/checker/regex/RegexAnnotatedTypeFactory.java

```diff
@@ -261,27 +418,121 @@ private int getRegexValue(AnnotationMirror anno) {
+        }
     }
 
     /**
      * Returns the group count value of the given annotation or 0 if there's a problem getting the
      * group count value.
+     *
+     * @param anno The annotation to extract group count from.
+     * @return Number of capturing groups as per the annotation.
      */
     public int getGroupCount(AnnotationMirror anno) {
-        AnnotationValue groupCountValue =
-                AnnotationUtils.getElementValuesWithDefaults(anno).get(regexValueElement);
-        // If group count value is null then there's no Regex annotation
-        // on the parameter so set the group count to 0. This would happen
-        // if a non-regex string is passed to Pattern.compile but warnings
-        // are suppressed.
-        return (groupCountValue == null) ? 0 : (Integer) groupCountValue.getValue();
+        if (anno.getAnnotationType().asElement().getSimpleName().contentEquals("Regex")) {
+            AnnotationValue groupCountValue =
+                    AnnotationUtils.getElementValuesWithDefaults(anno).get(regexValueElement);
+            // If group count value is null then there's no Regex annotation
+            // on the parameter so set the group count to 0. This would happen
+            // if a non-regex string is passed to Pattern.compile but warnings
+            // are suppressed.
+            if (groupCountValue != null) {
+                Object value = groupCountValue.getValue();
+                if (value instanceof Integer) return (Integer) value;
+            }
+        } else if (anno.getAnnotationType()
+                .asElement()
+                .getSimpleName()
+                .contentEquals("EnhancedRegex")) {
+            List<Integer> nonNullGroups =
+                    AnnotationUtils.getElementValueArray(anno, "value", Integer.class, false);
+            if (nonNullGroups.size() > 1) return nonNullGroups.get(nonNullGroups.size() - 1);
+            else throw new BugInCF("Size of the value array is less than 2.");
+        }
+        return 0;
+    }
+
+    /**
+     * Returns the list of non-null groups from the EnhancedRegex annotation and default if
+     * annotation is not present.
+     *
+     * @param anno The annotation from which the list is to be extracted.
+     * @return The extracted list of non-null groups.
+     */
+    public List<Integer> getNonNullGroups(AnnotationMirror anno) {
+        return AnnotationUtils.getElementValueArray(anno, "value", Integer.class, true);
     }
 
     /** Returns the number of groups in the given regex String. */
     public static int getGroupCount(@Regex String regexp) {
         return Pattern.compile(regexp).matcher("").groupCount();
     }
 
+    /**
+     * Returns the list of groups that are non-null in the given regex String.
+     *
+     * @param regexp The string to analyse.
+     * @return A list of non-null groups and the number of groups.
+     */
+    public static List<Integer> getNonNullGroups(@Regex String regexp) {
```

Comment:
This method is duplicative with `RegexUitl.getNonNullGroups`.  Don't duplicate code; instead, call the method in RegexUtil.

Reply:
I did it because `RegexUtil.getGroupCount(Pattern)` was declared private, so I assumed plume-lib would not allow anything other than `isRegex` and `asRegex` functions to be visible outside the RegexUtil class.

Reply:
Now that we have discovered a need for the method, I think it's fine to make the method public or protected.